src/global-environment.lisp

```diff
@@ -64,8 +64,13 @@
 (defun (setf var-declared-type) (new-value var)
```

Comment:
maybe we shouldn't overwrite at all if it's type=, so that nothing gets invalidated? (are things getting invalidateD?)

Reply:
Agree. Fixed